### PR TITLE
fix: expose experimental autoAlign input on icon button

### PR DESCRIPTION
**Pull request by Geoffrey0**
Closes carbon-design-system/carbon-components-angular#2855

The icon button did not yet expose the autoAlign input to enable the experimental floating-ui tooltip functionality. 

#### Changelog

**New**

* Added experimental autoAlign input on icon button component

### CODE REVIEW

**Review by Akshat55 — CHANGES_REQUESTED** [review]
Could you add a separate story for autoAlign? This way users can actually test it. You can reference the popover.stories.ts:

https://github.com/carbon-design-system/carbon-components-angular/blob/27e4e6c14e1aa1b0d3229a78e9f1a6dfbfcf96cf/src/popover/popover.stories.ts#L125

Since `autoalign` is not toggle-able for the time being, it's best to have a story showcasing the auto alignment.

**Comment by Geoffrey0** [comment]
Added the separate story indeed. Thanks for the feedback 🙏

File: src/button/icon-button.stories.ts [hunk 1]
```diff
@@ -97,3 +98,33 @@ Basic.args = {
...
+			[buttonAttributes]="buttonAttributes"
+			[disabled]="disabled"
+			[description]="description"
+			(click)="onClick($event)"
+			(mouseenter)="onMouseEnter($event)"
+			(mouseleave)="onMouseleave($event)"
+			(focus)="onFocus($event)"
+			(blur)="onBlur($event)">
+			<svg class="cds--btn__icon" cdsIcon="copy" size="16"></svg>
+		</cds-icon-button>
```
Comment (Akshat55) [inline src/button/icon-button.stories.ts:122]:
```suggestion
		<div style="height:3000px">
			Scrolling will update the position of the popover:
			<div style="position: absolute; top: 500px; left: 500px;">
				<cds-icon-button
					buttonId="icon-btn1"
					type="button"
					[kind]="kind"
					[size]="size"
					[align]="align"
					[isOpen]="isOpen"
					[autoAlign]="autoAlign"
					[buttonNgClass]="buttonNgClass"
					[buttonAttributes]="buttonAttributes"
					[disabled]="disabled"
					[description]="description"
					(click)="onClick($event)"
					(mouseenter)="onMouseEnter($event)"
					(mouseleave)="onMouseleave($event)"
					(focus)="onFocus($event)"
					(blur)="onBlur($event)">
					<svg class="cds--btn__icon" cdsIcon="copy" size="16"></svg>
				</cds-icon-button>
			</div>
		</div>
```

File: src/button/icon-button.stories.ts [hunk 2]
```diff
@@ -97,3 +98,33 @@ Basic.args = {
...
+	`
+});
+export const WithAutoAlign = AutoAlignTemplate.bind({});
+WithAutoAlign.args = {
+	autoAlign: true,
+	description: "Icon Description",
+	align: "top"
```
Comment (Akshat55) [inline src/button/icon-button.stories.ts:129]:
```suggestion
	align: "top",
	isOpen: true
```

**Review by Akshat55 — CHANGES_REQUESTED** [review]
We want to showcase that scrolling will change the position

**Comment by Akshat55** [comment]
Good to merge after those change

File: src/button/icon-button.stories.ts [hunk 3]
```diff
@@ -97,3 +98,39 @@ Basic.args = {
...
+		<div style="height:3000px">
+			Scrolling will update the position of the tooltip:
+			<div style="position: absolute; top: 500px; left: 500px;">
+				<cds-icon-button
+					buttonId="icon-btn1"
+					type="button"
+					[kind]="kind"
+					[size]="size"
+					[align]="align"
+					[autoAlign]="autoAlign"
```
Comment (Akshat55) [inline src/button/icon-button.stories.ts:114]:
```suggestion
					[autoAlign]="autoAlign"
					[isOpen]="isOpen"
```

**Review by Akshat55 — APPROVED** [review]

**Review by zvonimirfras — APPROVED** [review]

**Pull request merged**